test/functional/collection_tests.js

```diff
@@ -1137,439 +930,334 @@ describe('Collection', function() {
+      },
+
+      // The actual test we wish to run
+      test: function(done) {
+        db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, err => {
+          expect(err).to.not.exist;
 
-          db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, function(err) {
-            test.equal(null, err);
+          db.collection('shouldCorrectlyHandle0asIdForSave').save({ _id: 0 }, err => {
+            expect(err).to.not.exist;
             client.close();
             done();
           });
         });
-      });
-    }
-  });
-
-  /**
-   * @ignore
-   */
-  it('should correctly execute update with . field in selector', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+      }
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+    /**
+     * @ignore
+     */
+    it('should correctly execute update with . field in selector', {
+      metadata: {
+        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+      },
 
+      // The actual test we wish to run
+      test: function(done) {
         db
           .collection('executeUpdateWithElemMatch')
-          .update({ 'item.i': 1 }, { $set: { a: 1 } }, function(err) {
-            test.equal(null, err);
+          .update({ 'item.i': 1 }, { $set: { a: 1 } }, err => {
+            expect(err).to.not.exist;
 
             client.close();
             done();
           });
-      });
-    }
-  });
-
-  /**
-   * @ignore
-   */
-  it('should correctly execute update with elemMatch field in selector', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
+      }
+    });
 
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+    /**
+     * @ignore
+     */
+    it('should correctly execute update with elemMatch field in selector', {
+      metadata: {
+        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+      },
 
+      // The actual test we wish to run
+      test: function(done) {
         db
           .collection('executeUpdateWithElemMatch')
-          .update({ item: { $elemMatch: { name: 'my_name' } } }, { $set: { a: 1 } }, function(err) {
-            test.equal(null, err);
+          .update({ item: { $elemMatch: { name: 'my_name' } } }, { $set: { a: 1 } }, err => {
```

Comment:
I checked that the number of operations in the result equals 0, and the test is currently passing. Should I change the collection so that the number of operations is greater than 0?